mkfs/rootdir.c

```diff
@@ -362,6 +362,76 @@ static int add_symbolic_link(struct btrfs_trans_handle *trans,
 	return ret;
 }
 
+/*
+ * keep our extent size at 1MB max, this makes it easier to work
+ * inside the tiny block groups created during mkfs
+ */
+#define MAX_EXTENT_SIZE SZ_1M
+
+static int read_and_write_extent(struct btrfs_trans_handle *trans,
```

Comment:
I found the function name `read_and_write_extent()` a little hard to grasp.

The caller is `add_file_items()`, what about just a simple `add_one_file_item()`.

Reply:
Because it's not adding a file item, it's adding an extent.

I'll rename it to add_file_item_extent.